File: samples/ble_util/src/com/idevicesinc/sweetblue/ble_util/AlertManager.java
package com.idevicesinc.sweetblue.ble_util;

import com.idevicesinc.sweetblue.BleManager;
import com.idevicesinc.sweetblue.UhOh;

import android.app.AlertDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.DialogInterface.OnClickListener;
import android.view.Window;

/**
 * 
 * @author dougkoellmer
 */
public class AlertManager implements BleManager.UhOhListener
{
	private final Context m_context;
	private final BleManager m_bleMngr;
	
	public AlertManager(Context context, BleManager bleMngr)
	{
		m_context = context;
		m_bleMngr = bleMngr;
		
		m_bleMngr.setListener_UhOh(this);
	}
	
	public void showBleNotSupported()
	{
		AlertDialog.Builder builder = new AlertDialog.Builder(m_context);
		final AlertDialog dialog = builder.create();
		dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
		
		String dismiss = m_context.getResources().getString(R.string.generic_ok);
		String message = m_context.getResources().getString(R.string.ble_not_supported);

		OnClickListener clickListener = new OnClickListener()
		{
			@Override
			public void onClick(DialogInterface dialog, int which)
			{
				dialog.dismiss();
			}
		};
		
		dialog.setMessage(message);
		dialog.setButton(DialogInterface.BUTTON_NEUTRAL, dismiss, clickListener);
		dialog.show();
	}

	@Override public void onUhOh(BleManager manager, UhOh reason)
	{
		AlertDialog.Builder builder = new AlertDialog.Builder(m_context);
		final AlertDialog dialog = builder.create();

		OnClickListener clickListener = new OnClickListener()
		{
			@Override
			public void onClick(DialogInterface dialog, int which)
			{
				dialog.dismiss();
				
				if( which == DialogInterface.BUTTON_POSITIVE )
				{
					m_bleMngr.dropTacticalNuke();
				}
			}
		};
		
		String title = m_context.getResources().getString(R.string.uhoh_title);
		title = title.replace("{{reason}}", reason.name());
		dialog.setTitle(title);
		
		if( reason.getRemedy() == UhOh.Remedy.NUKE )
		{
			dialog.setMessage(m_context.getResources().getString(R.string.uhoh_message_nuke));
			dialog.setButton(DialogInterface.BUTTON_POSITIVE, m_context.getResources().getString(R.string.uhoh_message_nuke_drop), clickListener);
			dialog.setButton(DialogInterface.BUTTON_NEGATIVE, m_context.getResources().getString(R.string.uhoh_message_nuke_cancel), clickListener);
		}
		else if( reason.getRemedy() == UhOh.Remedy.RESTART_PHONE )
		{
			dialog.setMessage(m_context.getResources().getString(R.string.uhoh_message_phone_restart));
			dialog.setButton(DialogInterface.BUTTON_NEUTRAL, m_context.getResources().getString(R.string.uhoh_message_phone_restart_ok), clickListener);
		}
		else if( reason.getRemedy() == UhOh.Remedy.WAIT_AND_SEE )
		{
			dialog.setMessage(m_context.getResources().getString(R.string.uhoh_message_weirdness));
			dialog.setButton(DialogInterface.BUTTON_NEUTRAL, m_context.getResources().getString(R.string.uhoh_message_weirdness_ok), clickListener);
		}
		
		dialog.show();
	}
}

File: xamarin/SweetBlue/component/samples/Android/BleUtil/BleUtil/AlertManager.cs
using System;
using Android.Content;
using Android.App;
using Android.Views;
using Idevices.Sweetblue;

namespace BleUtil
{
	public class AlertManager : Java.Lang.Object
	{

		private Context m_context;
		public BleManager m_bleMgr;

		public AlertManager (Context context, BleManager mgr)
		{
			m_context = context;
			m_bleMgr = mgr;
			m_bleMgr.UhOh += (sender, e) => {
				handleUhOh(e.UhOhEvent);
			};
		}

		public void ShowBleNotSupported()
		{
			AlertDialog.Builder builder = new AlertDialog.Builder (m_context);

			String dismiss = m_context.GetString (Resource.String.generic_ok);
			String message = m_context.GetString (Resource.String.ble_not_supported);

			builder.SetMessage (message);
			builder.SetNeutralButton (dismiss, closeClick);
			AlertDialog dialog = builder.Create();
			dialog.RequestWindowFeature ((int) WindowFeatures.NoTitle);
			dialog.Show ();
		}

		private void handleUhOh(BleManager.UhOhListenerUhOhEvent ev)
		{
			AlertDialog.Builder builder = new AlertDialog.Builder (m_context);


			String title = m_context.GetString (Resource.String.uhoh_title);
			title = title.Replace ("{{reason}}", ev.UhOh ().Name ());
			builder.SetTitle (title);

			if (ev.Remedy () == BleManager.UhOhListenerRemedy.ResetBle) 
			{
				builder.SetMessage (m_context.GetString (Resource.String.uhoh_message_nuke));
				builder.SetPositiveButton (m_context.GetString (Resource.String.uhoh_message_nuke_drop), resetClick);
				builder.SetNegativeButton (m_context.GetString (Resource.String.uhoh_message_nuke_cancel), closeClick);
			}
			else if (ev.Remedy () == BleManager.UhOhListenerRemedy.RestartPhone) 
			{
				builder.SetMessage (m_context.GetString (Resource.String.uhoh_message_phone_restart));
				builder.SetNeutralButton (m_context.GetString (Resource.String.uhoh_message_phone_restart_ok), closeClick);
			}
			else if (ev.Remedy () == BleManager.UhOhListenerRemedy.WaitAndSee) 
			{
				builder.SetMessage (m_context.GetString (Resource.String.uhoh_message_weirdness));
				builder.SetNeutralButton (m_context.GetString (Resource.String.uhoh_message_weirdness_ok), closeClick);
			}
			AlertDialog dialog = builder.Create ();
			dialog.Show();
		}

		private void closeClick(object dialogObj, DialogClickEventArgs args)
		{			
			AlertDialog dialog = (AlertDialog)dialogObj;
			dialog.Dismiss ();
		}

		private void resetClick(object dialogObj, DialogClickEventArgs args)
		{
			AlertDialog dialog = (AlertDialog)dialogObj;
			dialog.Dismiss ();
			m_bleMgr.Reset ();
		}


	}
}

